tests/kernels/test_sphere_heat_kernel.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import torch
+import lab as B
+
+import geometric_kernels.torch
+from geometric_kernels.spaces.hypersphere import Hypersphere
+from geometric_kernels.kernels.geometric_kernels import MaternKarhunenLoeveKernel
+
+
+_TRUNCATION_LEVEL = 10
+_NU = 2.5
+
+
+def manifold_laplacian(manifold, x, egrad, ehess):
+    r"""
+    Computes the manifold Laplacian of a given function at a given point x.
+
+    :param manifold: manifold space, based on geomstats
+    :param x: point on the manifold at which to compute the Laplacian
+    :param egrad: Euclidean gradient of the function
+    :param ehess: Euclidean Hessian of the function
+
+    :return: manifold Laplacian
+    """
+    dim = manifold.dim
+    onb = torch.tensor(tangent_onb(manifold, x.detach().numpy()))
+    result = 0.
+    for j in range(dim):
+        cur_vec = onb[:, j]
+        egrad_x = egrad(x).detach().numpy()
+        ehess_x = ehess(x, cur_vec).detach().numpy()
+        hess_vec_prod = manifold.ehess2rhess(x.detach().numpy(), egrad_x, ehess_x, cur_vec.detach().numpy())
+        result += manifold.metric.inner_product(hess_vec_prod, cur_vec.detach().numpy(), base_point=x.detach().numpy())
+
+    return result
+
+
+def tangent_onb(manifold, x):
+    r"""
+    Computes an orthonormal basis on the tangent space at x.
+
+    :param manifold: manifold space, based on geomstats
+    :param x: point on the manifold
+
+    :return: [num, num] array containing the orthonormal basis
+    """
+    ambient_dim = manifold.dim + 1
+    manifold_dim = manifold.dim
+    ambient_onb = np.eye(ambient_dim)
+
+    projected_onb = np.ones_like(ambient_onb)
+
+    for j in range(ambient_dim):
+        cur_vec = ambient_onb[j, :]
+        projected_onb[j, :] = manifold.to_tangent(cur_vec, base_point=x)
```

Comment:
Can it be vectorized? Although it's a minor point

Reply:
I am not sure to understand, do you mean done in a batch manner, or something like this?

Reply:
What I Understood is that we may check if `manifold.to_tangent()` might handle an array of vectors, if so, then we may do this in a single shot.

Reply:
We indeed can, I just updated this.